nodejs/awsx/apigateway/requestValidator.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These APIs are currently experimental and may change.
+
+export type RequestValidator = "ALL" | "PARAMS_ONLY" | "BODY_ONLY";
+
+export interface Validators {
+    [name: string]: {
```

Comment:
what the key (i.e. 'name') here supposed to represent?

Reply:
You give the validators any names you want and then define the validator equal to one of those names. Since this isnt something users need to be bothered w/ I have moved it inline w the SwaggerSpec definition & changed name to validatorName for clarity.